app/Http/Controllers/BookingRequestController.php

```diff
@@ -154,4 +156,25 @@ public function destroy(BookingRequest $booking)
 
         return redirect(route('bookings.index'));
     }
+
+    public function downloadReferenceFiles($folder) 
+    {
+        $path = Storage::disk('public')->path($folder);
+
+        $zip = new ZipArchive;
+   
+        $fileName = $folder . '.zip';
+   
+        $zip->open(Storage::disk('public')->path($fileName), ZipArchive::CREATE);
+        $files = File::files($path);
+                              
+        foreach ($files as $key => $value) {
+            $relativeNameInZipFile = basename($value);
+            $zip->addFile($value, $relativeNameInZipFile);
+        }             
+        $zip->close();
```

Comment:
There is a code smeel here reported by [sonarcloud](https://sonarcloud.io/project/issues?id=CSU-Booking-Platform_application&open=AXcvi6nZOnzTSjnkyuLE&pullRequest=328&resolved=false&types=CODE_SMELL). the $key variable is unused.

Maybe you could do `foreach($files as $file){
... } ` ?